tests/test_agent.py

```
@@ -30,6 +35,107 @@ async def test_agent_close(self, agent: Agent) -> None:
+        assert "[No LLM loaded]" in response.content
+        assert "Hello" in response.content
+
+
+class TestAgentWithLLM:
+    """Tests for the Agent class with LLM integration"""
+
+    @pytest.fixture
+    def llm_config(self) -> AgentConfig:
+        return AgentConfig(
+            llm=LLMConfig(
+                model_path="/tmp/test-model.gguf",
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/test-model.gguf"

(S108)